x-pack/plugins/security_solution/server/lists_integration/endpoint/validators/blocklist_validator.ts

```diff
@@ -0,0 +1,284 @@
+  schema.literal('process.hash.sha256'),
+]);
+const ProcessExecutablePath = schema.literal('process.executable.caseless');
+const ProcessCodeSigner = schema.literal('process.Ext.code_signature');
+
+const ConditionEntryTypeSchema = schema.literal('match_any');
+const ConditionEntryOperatorSchema = schema.literal('included');
+
+type ConditionEntryFieldAllowedType =
+  | TypeOf<typeof ProcessHashField>
+  | TypeOf<typeof ProcessExecutablePath>
+  | TypeOf<typeof ProcessCodeSigner>;
+
+type BlocklistConditionEntry =
+  | {
+      field: ConditionEntryFieldAllowedType;
+      type: 'match_any';
+      operator: 'included';
+      value: string[];
+    }
+  | TypeOf<typeof WindowsSignerEntrySchema>;
+
+/*
+ * A generic Entry schema to be used for a specific entry schema depending on the OS
+ */
+const CommonEntrySchema = {
+  field: schema.oneOf([ProcessHashField, ProcessExecutablePath]),
+  type: ConditionEntryTypeSchema,
+  operator: ConditionEntryOperatorSchema,
+  // If field === HASH then validate hash with custom method, else validate string with minLength = 1
+  value: schema.conditional(
+    schema.siblingRef('field'),
+    ProcessHashField,
+    schema.arrayOf(
+      schema.string({
+        validate: (hash: string) =>
+          isValidHash(hash) ? undefined : `invalid hash value [${hash}]`,
+      })
+    ),
+    schema.conditional(
+      schema.siblingRef('field'),
+      ProcessExecutablePath,
+      schema.arrayOf(
+        schema.string({
+          validate: (pathValue: string) =>
+            pathValue.length > 0 ? undefined : `invalid path value [${pathValue}]`,
+        })
+      ),
+      schema.arrayOf(
+        schema.string({
+          validate: (signerValue: string) =>
+            signerValue.length > 0 ? undefined : `invalid signer value [${signerValue}]`,
+        })
+      )
+    )
+  ),
+};
+
+// Windows Signer entries use a Nested field that checks to ensure
+// that the certificate is trusted
+const WindowsSignerEntrySchema = schema.object({
+  type: schema.literal('nested'),
+  field: ProcessCodeSigner,
+  entries: schema.arrayOf(
+    schema.oneOf([
+      schema.object({
+        field: schema.literal('trusted'),
+        value: schema.literal('true'),
+        type: schema.literal('match'),
+        operator: schema.literal('included'),
+      }),
+      schema.object({
+        field: schema.literal('subject_name'),
+        value: schema.arrayOf(schema.string({ minLength: 1 })),
+        type: schema.literal('match_any'),
+        operator: schema.literal('included'),
+      }),
+    ]),
+    { minSize: 2, maxSize: 2 }
+  ),
+});
+
+const WindowsEntrySchema = schema.oneOf([
+  WindowsSignerEntrySchema,
+  schema.object({
+    ...CommonEntrySchema,
+    field: schema.oneOf([ProcessHashField, ProcessExecutablePath]),
+  }),
+]);
+
+const LinuxEntrySchema = schema.object({
+  ...CommonEntrySchema,
+});
+
+const MacEntrySchema = schema.object({
+  ...CommonEntrySchema,
+});
```

Comment:
We should just use one and call it `UnixSchema` or `MacOrLinuxEntrySchema` or something like that.